C/common/management_client.cpp

```diff
@@ -864,3 +867,308 @@ bool ManagementClient::addAuditEntry(const std::string& code,
 	}
 	return false;
 }
+
+/**
+ * Checks and validate the JWT bearer token coming from HTTP request
+ *
+ * @param request	HTTP request object
+ * @param claims	Map to fill with JWT public token claims
+ * @return		True on success, false otherwise
+ */
+bool ManagementClient::verifyAccessBearerToken(shared_ptr<HttpServer::Request> request,
+						map<string, string>& claims)
+{
+	string bearerToken = getAccessBearerToken(request);
+	if (bearerToken.length() == 0)
+	{
+		m_logger->info("Access bearer token has empty value");
+		return false;
+	}
+	return verifyBearerToken(bearerToken, claims);
+}
+
+/**
+ * Checks and validate the JWT bearer token string
+ *
+ * @param bearerToken	The bearer token string
+ * @param claims	Map to fill with JWT public token claims
+ * @return		True on success, false otherwise
+ */
+bool ManagementClient::verifyBearerToken(const string& bearer_token,
+					map<string, string>& claims)
+{
+	if (bearer_token.length() == 0)
+	{
+		m_logger->info("Bearer token has empty value");
+		claims.clear();
+		return false;
+	}
+
+	bool ret = true;
+
+	// Check token already exists in cache:
+	std::map<std::string, std::string>::iterator item;
+	// Acquire lock
+	m_mtx_rTokens.lock();
+
+	item = m_received_tokens.find(bearer_token);
+	if (item  == m_received_tokens.end())
+	{
+		bool verified = false;
+		// Token does not exist:
+		// Verify it by calling Fledge management endpoint
+		string url = "/fledge/service/verify_token";
+                string payload;
+                SimpleWeb::CaseInsensitiveMultimap header;
+                header.emplace("Authorization", "Bearer " + bearer_token);
+                auto res = this->getHttpClient()->request("POST", url.c_str(), payload, header);
+		Document doc;
+		string response = res->content.string();
+		doc.Parse(response.c_str());
+		if (doc.HasParseError())
+		{
+			bool httpError = (isdigit(response[0]) &&
+					  isdigit(response[1]) &&
+					  isdigit(response[2]) &&
+					  response[3]==':');
+			m_logger->error("%s error in service token verification: %s\n", 
+					httpError?"HTTP error during":"Failed to parse result of", 
+					response.c_str());
+			verified = false;
+			claims.clear();
+		}
+		else
+		{
+			if (doc.HasMember("error"))
+			{
+				if (doc["error"].IsString())
+				{
+					string error = doc["error"].GetString();
+					m_logger->error("Failed to parse token verification result, error %s",
+							error.c_str());
+				}
+				else
+				{
+					m_logger->error("Failed to parse token verification result: %s",
+							response.c_str());
+				}
+				claims.clear();
+				verified = false;
+			}
+			else
+			{
+				verified = true;
+
+				// Set token claims in the input map
+				if (doc["aud"].IsString() &&
```

Comment:
If one of these is missing or not a string the code will abort - are you sure tis can never happen?